### PR TITLE
APEXMALHAR-2380 Add MutablePair for Kinensis Operator for Recovery State

### PR DESCRIPTION
**Pull request by deepak-narkhede**
Add MutablePair for Kinensis Operator for Recovery State which will avoid default serialization caused by KinesisPair.

Tested with Kinesis to S3 app.

### CODE REVIEW

File: contrib/src/main/java/com/datatorrent/contrib/kinesis/AbstractKinesisInputOperator.java [hunk 1]
```diff
@@ -58,16 +32,27 @@
...
+import java.io.IOException;
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
```
Comment (chaithu14) [inline contrib/src/main/java/com/datatorrent/contrib/kinesis/AbstractKinesisInputOperator.java]: Why are you changing the import order?

Comment (deepak-narkhede) [reply contrib/src/main/java/com/datatorrent/contrib/kinesis/AbstractKinesisInputOperator.java]: @chaithu14 . It was not intentional, IDE did some import optimizations. I've reverted them.

File: contrib/src/main/java/com/datatorrent/contrib/kinesis/AbstractKinesisInputOperator.java [hunk 2]
```diff
@@ -565,11 +554,11 @@ public void emitTuples()
       outputPort.emit(tuple);
       if(!currentWindowRecoveryState.containsKey(shardId))
       {
-        currentWindowRecoveryState.put(shardId, new KinesisPair<String, Integer>(recordId, 1));
+        currentWindowRecoveryState.put(shardId, new MutablePair<String, Integer>(recordId, 1));
```
Comment (chaithu14) [inline contrib/src/main/java/com/datatorrent/contrib/kinesis/AbstractKinesisInputOperator.java]: I would suggest optimize this if block as below:
```java
MutablePair<String, Integer> shardOffsetAndCount = currentWindowRecoveryState.get(shardId);
if (shardOffsetAndCount == null) {
  currentWindowRecoveryState.put(shardId, new MutablePair<String, Integer>(recordId, 1));
} else {
  shardOffsetAndCount.setRight(shardOffsetAndCount.right+1);
}
```

Comment (deepak-narkhede) [reply contrib/src/main/java/com/datatorrent/contrib/kinesis/AbstractKinesisInputOperator.java]: @chaithu14 . Thanks, Done.

**Comment by chaithu14** [comment]
Looks okay to me.

**Pull request merged**